app/helpers/checkout_helper.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# CheckoutHelper
+module CheckoutHelper
+  def try_nil_or_invalid?(object)
+    object.nil? || object.invalid?
+  end
+
+  def valid_step
+    case true
+    when valid_address? then :address
+    when try_nil_or_invalid?(current_user.orders.last.delivery) then :delivery
```

Comment:
`when current_user.orders.last.delivery.try(:invalid?) then :delivery` which can be changed to:
`when current_order.delivery.try(:invalid?) then :delivery`
`current_user.orders.last` is duplicated a lot and can be moved to a helper method